wfe/wfe.go

```diff
@@ -2565,3 +2587,119 @@ func (wfe *WebFrontEndImpl) processRevocation(
 	})
 	return nil
 }
+
+// Verify the External Account Binding in the request and return the key ID if requested and found.
+func (wfe *WebFrontEndImpl) verifyEAB(newAcctReq newAccountRequest, outerPostData *authenticatedPOST) (*string, *acme.ProblemDetails) {
+	if newAcctReq.ExternalAccountBinding == nil {
+		if wfe.requireEAB {
+			return nil, acme.ExternalAccountRequiredProblem(
+				"The request must include a value for the 'externalAccountBinding' field")
+		}
+
+		return nil, nil
+	}
+
+	eabBytes, err := json.Marshal(newAcctReq.ExternalAccountBinding)
+	if err != nil {
+		return nil, acme.MalformedProblem(err.Error())
+	}
+
+	//1.  Verify that the value of the field is a well-formed JWS
+	innerJWS, err := wfe.parseJWS(string(eabBytes))
+	if err != nil {
+		return nil, acme.MalformedProblem(err.Error())
+	}
+
+	//2.  Verify that the JWS protected field meets the following criteria
+	//-  The "alg" field MUST indicate a MAC-based algorithm
+	//-  The "kid" field MUST contain the key identifier provided by the CA
+	//-  The "nonce" field MUST NOT be present
+	//-  The "url" field MUST be set to the same value as the outer JWS
+	keyID, prob := wfe.verifyEABPayloadHeader(innerJWS, outerPostData)
+	if prob != nil {
+		return nil, prob
+	}
+
+	//3.  Retrieve the MAC key corresponding to the key identifier in the
+	//    "kid" field
+	key, ok := wfe.db.GetExtenalAccountKeyByKeyID(keyID)
+	if !ok {
+		return nil, acme.BadPublicKeyProblem(
+			"the field 'kid' references a key that does not exist")
+	}
+
+	keyBytes, err := base64.RawURLEncoding.DecodeString(key)
+	if err != nil {
+		return nil, acme.MalformedProblem("failed to decode CA key")
+	}
+
+	//4.  Verify that the MAC on the JWS verifies using that MAC key
+	payload, err := innerJWS.Verify(keyBytes)
+	if err != nil {
+		return nil, acme.MalformedProblem(
+			fmt.Sprintf("external account binding JWS verification error: %s", err))
+	}
+
+	//5.  Verify that the payload of the JWS represents the same key as was
+	//    used to verify the outer JWS (i.e., the "jwk" field of the outer
+	//    JWS)
+	prob = wfe.verifyEABMatchesKey(payload, outerPostData.jwk)
+	if prob != nil {
+		return nil, prob
+	}
+
+	wfe.log.Printf("External Account Binding with CA using kid %q", keyID)
```

Comment:
```suggestion
	wfe.log.Printf("Successful newAccount External Account Binding with CA using kid %q", keyID)
```

I think a bit more context here would help.